File: farm/farm/src/main/java/com/glauco/farm/api/services/talhao/TalhaoService.java
package com.glauco.farm.api.services.talhao;

import org.springframework.data.domain.Page;
import org.springframework.stereotype.Component;

import com.glauco.farm.api.model.Talhao;
import com.glauco.farm.api.services.BaseService;

@Component
public interface TalhaoService extends BaseService<Talhao> {

    Page<Talhao> findByFazendaId(int page, int count, String fazendaId);
}

File: farm/farm-front/src/app/services/talhao/talhao.service.ts
import { Injectable } from '@angular/core';
import {BaseService} from '../base.service';
import {HttpClient} from '@angular/common/http';
import {API_PATH} from '../../app.const';

@Injectable()
export class TalhaoService extends BaseService {

  constructor(http: HttpClient) {
    super(http, '/api/talhao');
  }

  findByFazendaId(page: number, count: number, fazendaId: string) {
    return this.http.get(`${API_PATH}/${this.path}/${page}/${count}/${fazendaId}`);
  }
}
